overlord/restart/restart.go

```diff
@@ -162,15 +162,15 @@ func (m *RestartManager) StartUp() error {
 				continue
 			}
 
-			logger.Debugf("system restart happened, mark as done task %q for change %s", t.Summary(), chg.ID())
-			t.SetStatus(state.DoneStatus)
+			waitStatus := t.WaitStatus()
+			logger.Debugf("system restart happened, marking task %q as %s for change %s", t.Summary(), waitStatus, chg.ID())
+			t.SetStatus(waitStatus)
```

Comment:
this change is not really tested as the tests always use DoneStatus atm

Reply:
I added a case in the restart tests to verify another status than DoneStatus